drivers/FlashIAP.h

```diff
@@ -63,8 +63,10 @@ namespace mbed {
  */
 class FlashIAP : private NonCopyable<FlashIAP> {
 public:
-    FlashIAP();
-    ~FlashIAP();
+    constexpr FlashIAP() : _flash(), _page_buf(nullptr)
+    {
+
+    }
```

Comment:
Just going to mention in passing that C++11 gives you another option here, which is kind of neater, but maybe less familiar.

You could do

    // no need to declare constructor

    private:
        flash_t _flash{};
        uint8_t *_page_buf = nullptr;
  
C++ Core Guidelines recommends that [(see C.48 and C.49)](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Rc-in-class-initializer), but obviously that syntax is unfamiliar in our tree, what with us only just switching to C++11/14.

Reply:
I think we should first update our guidelines and then do this for the codebase (good mention !)